### DISCUSSION TITLE
Trying to upload a site to Pages, but it isn't uploading the correct page.

### DISCUSSION THREAD

**Discussion by mattmoon00** [community/community#42434] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I created a react app using next.js and then uploaded all that code to my repository. The code is all correct on my repository, but when I try to publish it using Pages, it shows the starter page as if my next.js app hasn't been edited at all. I have no clue why this is happening but I've tried everything I can think of, including deleting the repository and re-uploading it and trying to deploy again, and it is still giving me the same issue.

**Comment by getlost01** [comment]
In order to use GitHub Pages to host your Next.js app, you'll need to ensure that your repository meets the following requirements:

1. Your repository must contain a branch named` gh-pages`. This branch will be used to host your website.
2. Your repository must contain a file named `index.html` at the root of the `gh-pages` branch. This file will be the default page that is shown when someone visits your website.
3. Your repository must be configured to use GitHub Pages. To do this, go to the repository's settings page and scroll down to the "GitHub Pages" section. From there, you can choose the `gh-pages` branch as the source for your GitHub Pages website.

Try and check is your page is working with these solutions or not.

**Reply by mattmoon00** [reply]
I realized it was showing the readme.md file. I followed these steps but it's still showing the readme on the deployment.

**Reply by mattmoon00** [reply]
Update: I added my code into the index.html file and it just shows text. I'm not sure how to get my react code to load if the file being loaded is a .html file.

**Reply by getlost01** [reply]
Since I found your way to import images is wrong due to which your app is not built right in vercel, I have reviewed and updated your code and also sent a pull request, to merge It.

Here I also upload it to my vercel, check it out.
https://matts-portfolio-website-9778.vercel.app/

**Reply by getlost01** [reply]
@mattmoon00  is your issue resolved by your side problem? Is vercel works properly?

**Comment by jfullstackdev** [comment]
there are three deployments 

![image](https://user-images.githubusercontent.com/47092464/209420970-e903ef22-ae25-46ef-b038-b597df3ac709.png)

![image](https://user-images.githubusercontent.com/47092464/209420984-306f059d-1558-4ae2-ac72-f87465bedd4f.png)

![image](https://user-images.githubusercontent.com/47092464/209421001-81d01b48-8f93-4cac-870e-6b3ef8076141.png)

they are all the same, and they are not GitHub Pages deployment

did you deploy it at the Settings of the repo ? deploy it using your default branch because you don't have `gh-pages`
